packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -931,6 +945,49 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     await this._txPool.updatePendingAndQueued();
   }
 
+  public async setAccountBalance(
+    address: Address,
+    newBalance: BN
+  ): Promise<void> {
+    const account = await this._stateManager.getAccount(address);
+    account.balance = newBalance;
+    await this._stateManager.putAccount(address, account);
+    await this._persistIrregularWorldState();
+  }
+
+  public async setAccountCode(
+    address: Address,
+    newCode: Buffer
+  ): Promise<void> {
+    await this._stateManager.putContractCode(address, newCode);
+    await this._persistIrregularWorldState();
+  }
+
+  public async setAccountNonce(address: Address, newNonce: BN): Promise<void> {
+    const account = await this._stateManager.getAccount(address);
```

Comment:
I haven't think about this until now, but what should happen if there are pending txs?

If automining is disabled there's both a nonce that comes from the state manager, but there may also be pending txs that each of them have nonces.

Not sure what should we do here. Drop the pending txs if their nonces are too low? Forbid this method if there are pending txs?

Wdyt, @feuGeneA @fvictorio ?

Reply:
I thought more about this, and I think we should forbid using this method if you have pending transactions. I think it's highly improbable that someone will be limited by this in practice, and it significantly simplifies the implementation.

Reply:
That seems fine to me, yes :+1:

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1435/commits/74ecff3a60db7c360aebc8ed9853ce932af6bd1b

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1435/commits/8e9a113376462500a2ad6ade6908a6680fd0a59b

I took a very quick look at actually checking by address, and it didn't seem trivial so I skipped it for now, but I'm happy to look into that if you guys think it's important.

Reply:
I'm fine with this. We can always relax that restriction by checking by address.